tx/builder.go

```diff
@@ -7,76 +7,105 @@ package tx
 
 import (
 	"encoding/binary"
+	"math/big"
 
 	"github.com/vechain/thor/v2/thor"
 )
 
 // Builder to make it easy to build transaction.
 type Builder struct {
-	body body
+	legacyTx     LegacyTransaction
+	dynamicFeeTx DynamicFeeTransaction
 }
```

Comment:
What if there was `Builder`, `LegacyBuilder` and `DynamicFeeBuilder`, where the Builder would point to the `DynamicFeeBuilder` ?

Reply:
added two separate builders